sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -214,11 +214,11 @@ public static ServiceBusReceivedMessage AmqpMessageToSBMessage(AmqpMessage amqpM
 
             if ((amqpMessage.BodyType & SectionFlag.Data) != 0 && amqpMessage.DataBody != null)
             {
-                annotatedMessage = new AmqpAnnotatedMessage(amqpMessage.GetDataViaDataBody());
+                annotatedMessage = new AmqpAnnotatedMessage(new AmqpMessageBody(amqpMessage.GetDataViaDataBody()));
             }
             else
             {
-                annotatedMessage = new AmqpAnnotatedMessage(new ReadOnlyMemory<byte>[] { Array.Empty<byte>() });
+                annotatedMessage = new AmqpAnnotatedMessage(new AmqpMessageBody(new ReadOnlyMemory<byte>[] { Array.Empty<byte>() }));
```

Comment:
Should we consider having a `static readonly` for the empty memory array rather than allocating a new enumerable for each call? Since the output is a read-only sequence of a read-only memory, is there any benefit to a unique instance when its empty?